src/Orleans/Runtime/IGrainLifeCycle.cs

```diff
@@ -0,0 +1,26 @@
+﻿
+namespace Orleans.Runtime
+{
+    /// <summary>
+    /// Stages of a grains lifecycle.
+    /// TODO: Add more later, see ActivationInitializationStage
+    /// Full grain lifecycle, including register, state setup, and 
+    ///   stream cleanup should all eventually be triggerd by the 
+    ///   grian lifecycle.
+    /// </summary>
+    public enum GrainLifecyleStage
+    {
+        //None,
+        //Register,
+        SetupState,  // Setup grain state prior to activation
+        //InvokeActivate,
+        //Completed
```

Comment:
Agreed.  I also suspect there will need to be internal stages and application layer stages (for instance, application probably shouldn't have access to the registration stage), but at this point we're just setting up the basic framework.  More work is needed before grain lifecycle advantages will be fully demonstrable.